src/client.rs

```diff
@@ -114,16 +114,25 @@ impl<'mpesa, Env: ApiEnvironment> Mpesa<Env> {
             .await?;
         if response.status().is_success() {
             let value = response.json::<Value>().await?;
+            let error_message ="Failed to extract token from the response";
             let access_token = value
                 .get("access_token")
-                .ok_or_else(|| MpesaError::AuthenticationError(value.clone()))?;
+                .ok_or_else(|| MpesaError::AuthenticationError(
+                    ApiError::new(value.to_string(),String::new(),error_message.to_string()
+                )))?;
             let access_token = access_token
                 .as_str()
-                .ok_or_else(|| MpesaError::AuthenticationError(value.clone()))?;
+                .ok_or_else(|| MpesaError::AuthenticationError(
+                    ApiError::new(value.to_string(), String::new(),error_message.to_string())
+                ))?;
```

Comment:
I don't think an `ApiError` would be best to represent an error where we could not get the `access_token` from a successful request or converting a `serde_json` value to a `str`. Try to see if you can come up with a better way to handle these errors